plugin/kubernetes/controller.go

```diff
@@ -368,27 +353,24 @@ func (dns *dnsControl) PodIndex(ip string) (pods []*object.Pod) {
 		if !ok {
 			continue
 		}
-		pods = append(pods, p)
+		return []*object.Pod{p}
```

Comment:
This index is unique, it will only return one pod, so just return the first pod object we find